Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryTransforms.java

```diff
@@ -0,0 +1,187 @@
+ *       and/or other materials provided with the distribution.
+ *     * Neither the name of the United States Government nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE UNITED STATES GOVERNMENT BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package gov.hhs.fha.nhinc.patientdiscovery.audit.transform;
+
+import com.services.nhinc.schema.auditmessage.AuditMessageType;
+import com.services.nhinc.schema.auditmessage.ParticipantObjectIdentificationType;
+import gov.hhs.fha.nhinc.audit.transform.AuditTransform;
+import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
+import gov.hhs.fha.nhinc.patientdiscovery.audit.PatientDiscoveryTransformConstants;
+import gov.hhs.fha.nhinc.transform.marshallers.JAXBContextHandler;
+import gov.hhs.fha.nhinc.util.HomeCommunityMap;
+import java.io.ByteArrayOutputStream;
+import java.util.List;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import org.apache.log4j.Logger;
+import org.hl7.v3.II;
+import org.hl7.v3.PRPAIN201305UV02;
+import org.hl7.v3.PRPAIN201306UV02;
+import org.hl7.v3.PRPAIN201306UV02MFMIMT700711UV01Subject1;
+
+/**
+ *
+ * @author achidamb
+ * @param <T>
+ * @param <K>
+ */
+public class PatientDiscoveryTransforms<T extends PRPAIN201305UV02, K extends PRPAIN201306UV02> extends AuditTransform<T, K> {
+
+    private static final Logger LOG = Logger.getLogger(PatientDiscoveryTransforms.class);
+
+    /**
+     *
+     * @param msg
+     * @param assertion
+     * @param auditMsg
+     * @param response
+     * @return
+     */
+    @Override
+    protected AuditMessageType getParticipantObjectIdentification(PRPAIN201305UV02 request, PRPAIN201306UV02 response, AssertionType assertion, AuditMessageType auditMsg) {
+        auditMsg = getPatientParticipantObjectIdentification(request, response, auditMsg);
+        try {
+            auditMsg = getQueryParamsParticipantObjectIdentification(request, response, auditMsg);
+        } catch (JAXBException ex) {
+            LOG.debug("Error while creating ParticipantObjectIdentificationQueryByParameters segemnt : " + ex);
```

Comment:
Typo, should be "segment" -- also, change this to LOG.error if it's an error, or change to LOG.warn("Unable to create ...." ... )

Also, since I'm already commenting here, just a reminder -- EVERYWHERE you log an exception, use ("msg" + ex.getLocalizedMessage(), ex)